src/main/java/com/yahoo/bullet/pubsub/rest/RESTSubscriber.java

```diff
@@ -72,7 +72,8 @@ public RESTSubscriber(int maxUncommittedMessages, List<String> urls, CloseableHt
                     HttpEntity httpEntity = response.getEntity();
                     String message = EntityUtils.toString(httpEntity, RESTPubSub.UTF_8);
                     log.debug("Received message from url: {}. Message was {}", url, message);
-                    messages.add(GSON.fromJson(message, PubSubMessage.class));
+                    PubSubMessage pubSubMessage = GSON.fromJson(message, PubSubMessage.class);
+                    messages.add(PubSubMessage.fromJSON(pubSubMessage));
```

Comment:
use the other PubSubMessage#fromJson(String)

Reply:
Can't since we need to deserialize using the GSON in this class (for RestMetadata). I think I'll rework this though so we can hook into the GSON in the PubSubMessage instead of allowing for deserialization separately and then fixing the PubSubMessage.